reana_server/rest/info.py

```diff
@@ -136,3 +192,8 @@ class InfoSchema(Schema):
     )
     default_workspace = fields.Dict(keys=fields.Str(), values=fields.Str())
     compute_backends = fields.Dict(keys=fields.Str(), values=fields.List(fields.Str()))
+    kubernetes_memory_limit = fields.Dict(keys=fields.Str(), values=fields.Str())
+    kubernetes_max_memory_limit = fields.Dict(keys=fields.Str(), values=fields.Str())
+    maximum_workspace_retention_period = fields.Dict(
+        keys=fields.Str(), values=fields.Str()
+    )
```

Comment:
I think this schema is wrong for `workspaces_available` and `compute_backends` given that the values are both strings (for `title`) and lists (for `value`). What do you think about this new schema?

```python
class ListStringInfoValue(Schema):
    """Schema for a value represented by a list of strings."""
    title = fields.String()
    value = fields.List(fields.String())


class StringInfoValue(Schema):
    """Schema for a value represented by a string."""
    title = fields.String()
    value = fields.String(allow_none=True)


class InfoSchema(Schema):
    """Marshmallow schema for ``info`` endpoint."""

    workspaces_available = fields.Nested(ListStringInfoValue)
    default_workspace = fields.Nested(StringInfoValue)
    compute_backends = fields.Nested(ListStringInfoValue)
    kubernetes_memory_limit = fields.Nested(StringInfoValue)
    kubernetes_max_memory_limit = fields.Nested(StringInfoValue)
    maximum_workspace_retention_period = fields.Nested(StringInfoValue)
```

I also found out that we can use this schema to automatically generate the OpenAPI specification!
In particular:

1. Add the Marshmallow plugin in `scripts/generate_openapi_spec.py`

    ```diff
    -        plugins=("apispec.ext.flask",),
    +        plugins=("apispec.ext.flask", "apispec.ext.marshmallow"),
    ```

2. Use `InfoSchema` in the OpenAPI specification

    ```yaml
    responses:
      200:
        description: >-
          Request succeeded. The response contains general info about the cluster.
        schema: InfoSchema
        examples:
          ...
    ```

Should we try this approach?

Reply:
You are right about schema. One thing from your example is that `default_workspace` is not nullable, but if this doesn't have a problem, we can use the schemas above.

I tried generating OpenAPI spec using schema. It looks pretty nice.

Reply:
> You are right about schema. One thing from your example is that `default_workspace` is not nullable, but if this doesn't have a problem, we can use the schemas above.

In this case we can use two different schemas, one with `allow_none=False` and the other with `allow_none=True`, if you prefer to be more precise!